Documentation/chapters/decyzje-projektowe.tex

```diff
@@ -0,0 +1,150 @@
+Na uczelni poznaliśmy 2 technologie tworzenia takich serwisów.
+
+\paragraph{Java}
+Java to język paradygmatu obiektowego, bardzo popularny w systemach bankowych.
+W trakcie zajęć na uczelni poznaliśmy sposób tworzenia serwisów HTTP w technologii Spring.
+Części naszego zespołu nie podobają się konwencje narzucone przez tą technologie, jak i sposób komunikacji z bazą danych.
+Należy zaznaczyć, że nikt z naszego zespołu nie posada komercyjnego doświadczenia z tą technologią.
+
+\paragraph{.NET}
+C\# to jeden z języków z rodziny .NET, który mielimy szanse poznać na uczelni.
+Łączy on w sobie paradygmaty obiektowe i funkcyjne, dzięki czemu jest bardziej elastyczny, jeśli chodzi o styl programowania.
+Trzy z czterech osób w naszym zespole ma doświadczenie komercyjne w tej technologi.
+
+\paragraph{Wybór}
+Z uwagi na doświadczenie zespołu postawiliśmy na C\#.
+
+\subsubsection{Baza danych}
+Na uczelni dogłębnie poznaliśmy bazę danych PostgreSql, zespół ma komercyjne doświadczenie z tą bazą, więc nie braliśmy innych rozwiązań pod uwagę.
+
+\subsection{Narzędzia programistyczne}
+\label{subsec:narzedzia-programistyczne}
+Nasz zespół składa się z osób, które mają znaczne doświadczenie w oprogramowaniu, którego używa na co dzień w pracy.
+Wybrany przez nas stos technologiczny składa się z .NET C\#, Angular oraz PostgreSql.
+Wszystkie te technologie, pomimo iż są utrzymywane przez ogromne korporacje, są otwarto-źródłowe, co w praktyce pozwala na tworzenie narzędzi deweloperskich przez niezależnych twórców, dzięki czemu nie bylibyśmy uwiązani do konkretnego rozwiązania.
+Wybór oprogramowania do tworzenia kodu został podyktowany wybraną technologią a w drugiej kolejności z uwagi na zróżnicowany sprzęt komputerowy, którym dysponujemy  dostępnością oprogramowania na wielu platformach (tj.: Windows, MacOS, Linux).
+
+\subsubsection{.NET/C\#}
+.NET to framework stworzony i rozwijany przez Microsoft wspólnie ze społecznością otwartego oprogramowania.
+Pierwszym oprogramowaniem, które przychodzi nam do głowy, gdy słyszymy .NET, jest Microsoft Visual Studio.
+Oprogramowanie to powstało w 1997 roku i przez wiele lat było konsekwentnie ulepszane.
+W 2016 roku VS zostało wydane na komputery Mac, jednakże, po 7 latach, w 2023 roku Microsoft ogłosił zakończenie wsparcia dla wersji MacOS.
+Jeśli chodzi o systemy Linux, to VS nigdy nie zostało wydane na ten system.
+
+Zespół projektowy bardzo nie chciał uzależnić pracy nad projektem od systemu Windows, Microsoft poleca użycie programu Visual Studio Code, który jest wspierany na wszystkich systemach operacyjnych, których używamy.
+Niestety VSC jest edytorem przeznaczenia ogólnego, co za tym idzie, znaczna większość specjalistycznych dla danej technologii narzędzi jest obsługiwana przy pomocy rozszerzeń.
+Rozszerzenia mogą tworzyć wszyscy, jest ich całkiem sporo niestety wszystkie rozszerzenia skupiające się na technologii .NET/C\# są we wczesnej fazie rozwoju i nie wspierają tak podstawowych rzeczy, jak debugowanie kodu.
```

Comment:
W VSC Można normalnie debugować i nie ma z tym problemu

Reply:
Konfiguracja tego rozszerzenia jest trudna, a jak już się uda to nie możesz podejrzeć i edytować pamięci na żywo, profilować call stack itd. Można to wsm wypisać ale nie wiem czy warto